shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm

```diff
@@ -83,13 +83,21 @@ - (void)application:(UIApplication*)application
 - (void)userNotificationCenter:(UNUserNotificationCenter*)center
        willPresentNotification:(UNNotification*)notification
          withCompletionHandler:
-             (void (^)(UNNotificationPresentationOptions options))completionHandler
-    API_AVAILABLE(ios(10)) {
-  if (@available(iOS 10.0, *)) {
-    [_lifeCycleDelegate userNotificationCenter:center
-                       willPresentNotification:notification
-                         withCompletionHandler:completionHandler];
-  }
+             (void (^)(UNNotificationPresentationOptions options))completionHandler {
+  [_lifeCycleDelegate userNotificationCenter:center
```

Comment:
This call should be gated by 'respondsToSelector:' since the method is optional.

Reply:
Would we need it here even if there's a `respondToSelector:` in [FlutterPluginAppLifeCycleDelegate](https://github.com/flutter/engine/pull/9864/files#diff-71f628ed3f0d51c80d58e053985f8a81R280)? This method should also only be called if the user deliberately sets the `FlutterAppLifeCycleProvider` (FlutterAppDelegate.mm) as the `UNUserNotificationDelegate`.

Reply:
Yes, it's still needed.  The FlutterPluginAppLifeCycleDelegate is a different layer.  For your purposes here all you need to consider is UNUserNotificationCenterDelegate.  If you look at the header for that protocol you'll see this method is optional.  Calling any optional method in a protocol should be guarded by a reflection test.

Reply:
That makes sense. done